Server/DiscordBot.cs

```diff
@@ -1,160 +1,274 @@
-                    } else if (msg.Content.StartsWith(mentionPrefix)) {
-                        await msg.Channel.TriggerTypingAsync();
-                        resp = string.Join('\n', CommandHandler.GetResult(msg.Content[mentionPrefix.Length..].TrimStart()).ReturnStrings);
-                    }
-                    if (resp != null)
+                else if (message.StartsWith(localSettings.Prefix))
+                {
+                    args = message[localSettings.Prefix.Length..];
+                }
+                else if (message.StartsWith($"<@{client!.CurrentUser.Id}>"))
+                {
+                    args = message[client!.CurrentUser.Mention.Length..].TrimStart();
+                }
+                if (args != null)
+                {
+                    await arg.Channel.TriggerTypingAsync();
+                    string resp = string.Join('\n', CommandHandler.GetResult(args).ReturnStrings);
+                    if (localSettings.LogCommands)
                     {
-                        foreach (string mesg in SplitMessage(resp))
-                            await msg.RespondAsync(mesg);
+                        logger.Info($"\"{arg.Author.Username}\" ran the command: \"{message}\" via discord");
                     }
-                } catch (Exception e) {
-                    Logger.Error(e);
+                    foreach (string mesg in SplitMessage(resp))
+                        await (arg as SocketUserMessage).ReplyAsync(mesg);
                 }
-            };
-            DiscordClient.ClientErrored += (_, args) => {
-                Logger.Error("Discord client caught an error in handler!");
-                Logger.Error(args.Exception);
-                return Task.CompletedTask;
-            };
-            DiscordClient.SocketErrored += (_, args) => {
-                Logger.Error("Discord client caught an error on socket!");
-                Logger.Error(args.Exception);
-                return Task.CompletedTask;
-            };
-        } catch (Exception e) {
-            Logger.Error("Exception occurred in discord runner!");
-            Logger.Error(e);
+            }
+            catch (Exception e)
+            {
+                logger.Error(e);
+            }
+        }
+        else
+        {
+            //don't respond to commands not in these channels, and no bots
+            //probably don't print out error message because DDOS
+        }
+    }
+
+    private void OnLoadSettings()
+    {
+        Settings.DiscordTable oldSettings = localSettings;
+        localSettings = Settings.Instance.Discord;
+
+        if (localSettings.CommandChannel == null)
+            logger.Warn("You probably should set your CommandChannel in settings.json");
+        if (localSettings.AdminChannel == null)
+            logger.Warn("You probably should set your AdminChannel in settings.json");
```

Comment:
Bug: when the `loadsettings` command is used this will wrongly warn the user even when the Discord bot is disabled.

```suggestion
```

This warning is not needed here anymore, because if Discord is enabled, the check in `Init()` will be called to print the new warning.